superset/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -90,6 +87,7 @@ export const exploreReducer = function (state, action) {
           chartUpdateEndTime: null,
           chartUpdateStartTime: now(),
           triggerQuery: false,
+          queryRequest: action.queryRequest,
```

Comment:
The problem with keeping latestQueryId in the state is that: when ajax returns before reducer updates the state with new latestQueryId, old queryResponse is dispatched, sort of like a race condition